blocks/procedures.js

```diff
@@ -537,8 +537,11 @@ Blockly.Blocks['procedures_mutatorarg'] = {
     // Hack: override showEditor to do just a little bit more work.
     // We don't have a good place to hook into the start of a text edit.
     field.oldShowEditorFn_ = field.showEditor_;
+    // TODO: Understand what this is in newShowEditorFn, and annotate properly.
     const newShowEditorFn = function() {
+      /* eslint-disable-next-line no-invalid-this */
       this.createdVariables_ = [];
+      /* eslint-disable-next-line no-invalid-this */
       this.oldShowEditorFn_();
     };
     field.showEditor_ = newShowEditorFn;
```

Comment:
My understanding is that because the function is invoked as a method, `this` should be bound to whatever object the function exists on ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#as_an_object_method)). So I think it should be `FieldTextInput` or whatever the particular derivative is. However, I don't think closure is going to like that because `oldShowEditorFn_` doesn't exist on `FieldTextInput`, only this kluged version of `FieldTextInput`.

Might be better to leave as is for now, and fix the hack by adding in a hook for the start of a text edit.

Reply:
I fixed the annotation based on that description.